### PR TITLE
KIALI-1103 Fix versions components in about and refactor

### PR DESCRIPTION
**Pull request by aljesusg**
Jira: [KIALI-1103](https://issues.jboss.org/browse/KIALI-1103)

- Moved about info to store-persist (Don't need to check everytime the versions)
- Removed authentication header in getStatus (Don't need it in /api and /api/status)
- Added comments to service/Api and refactor headers

### CODE REVIEW

File: src/services/Api.ts [hunk 1]
```diff
@@ -14,21 +14,38 @@ interface Response<T> {
...
+
 const loginHeaders = { 'X-Auth-Type-Kiali-UI': '1' };
-const authHeader = (auth: string) => {
-  return { Authorization: auth };
+
+const authHeader = (auth: string) => ({ Authorization: auth });
+
+/**  Helpers to Requests */
+
+const getHeaders = (auth: string | undefined = undefined) => {
```
Comment (jotak) [inline src/services/Api.ts]: arguments can be simplified: `(auth?: string)` , i think it works the same

File: src/services/Api.ts [hunk 2]
```diff
@@ -14,21 +14,38 @@ interface Response<T> {
...
   return { username: username, password: password };
 };
 
-let newRequest = <T>(method: string, url: string, queryParams: any, data: any, auth: string) => {
+let newRequest = <T>(
+  method: string,
+  url: string,
+  queryParams: any,
+  data: any,
+  auth: string | undefined = undefined
```
Comment (jotak) [inline src/services/Api.ts]: same here

File: src/services/Api.ts (hunk 1 above)
Comment (aljesusg) [reply src/services/Api.ts]: Yes, you are right

Comment (aljesusg) [reply src/services/Api.ts]: Changed

**Comment by jotak** [comment]
I'm not sure about moving the status to a store. Why wouldn't we want up-to-date information? Especially I see the "Kiali state" field that might be running or not, that could be a changing information that worth keeping up-to-date.

**Comment by aljesusg** [comment]
This was in store, we don't have any tool to check if the server is running only in the login step.

**Comment by aljesusg** [comment]
Maybe we can add in the masthead a status of Kiali server and refresh status every 5 seconds to show if Kiali is working. But the point of move this to store is avoid to request Everytime the versions of components when they won't change ^^

**Comment by jotak** [comment]
@aljesusg hmm, on server-side I saw something about a "running" status but maybe this is something unused then. See https://github.com/kiali/kiali/blob/master/status/status.go#L72
It's a bit annoying to see dead code such as this.

But ok then, +1 to merge. It will still be time to change if we're adding more information in status in the future.

**Comment by aljesusg** [comment]
Ok I'll open a JIRA to make the changes that we dicussed in the chat, about the dead code I don't know when change that ^^

**Pull request merged**